deployments/rialto/docker-compose.yml

```
@@ -215,3 +215,9 @@ services:
       - "4567:4567"
     depends_on:
       - grafana-dashboard
+  front-end:
+    build:
+      context: .
+      dockerfile: ./Front-end.Dockerfile
```

I think we should add `environment` here and by default set the values to `localhost`:
```
  environment:
      SUBSTRATE_PROVIDER:  ${UI_SUBSTRATE_PROVIDER:-ws://localhost:9933}
      ETHEREUM_PROVIDER: ${UI_ETHEREUM_PROVIDER:-http://localhost:8545}	      
      EXPECTED_ETHEREUM_NETWORK_ID: 105
```

---

This is beyond my knowledge :smile: 
From where would `${UI_SUBSTRATE_PROVIDER:-ws://localhost:9933}` get the `UI_SUBSTRATE_PROVIDER` variable? is the `-ws://localhost:9933` a default then?

---

So the `docker-compose` file can pass ENV variables, just like you would do (via CLI flags) when building `docker` image from CLI. The `environment` section will pass these ENV variables to the build, so they can be read by the `Dockerfile` that you created.

Secondly `docker-compose` supports `.env` files (again :exploding_head:), so when you run `docker-compose up` it's going to read `.env` file and substitute `${UI_SUBSTRATE_PROVIDER}` with a value found in environment. If that's not set we will fall back to default `ws://localhost:9933` value.

I request that, because the way we configure the rest of the whole environment (all services in `docker-compose`) is done via compose-specific `.env` file, so I expect the UI should be the same. By default we want it to just connect to `localhost` so that when you run `$ docker-compose up` the UI at `localhost:8080` would automatically connect to nodes that has been started at `:8545` and `:9933`.

Hope this `ENV` inception makes any sense :)

---

Wow, thanks a lot for the explanations. every time I touch such things it's :exploding_head:  for me, I get it.. and forget it a couple weeks later :roll_eyes: 
So in the end that should do the trick right?:
```suggestion
      dockerfile: ./Front-end.Dockerfile
      environment:
        SUBSTRATE_PROVIDER:  ${UI_SUBSTRATE_PROVIDER:-ws://localhost:9933}
        ETHEREUM_PROVIDER: ${UI_ETHEREUM_PROVIDER:-http://localhost:8545}	      
        EXPECTED_ETHEREUM_NETWORK_ID: 105
```

---

Yup!